### PR TITLE
Document increase in memory usage due to mempool/dbcache sharing

### PR DESCRIPTION
**Pull request by TheBlueMatt**

### CODE REVIEW

**Comment by TheBlueMatt** [comment]
I believe this is a sufficient fix for #9857.

**Comment by paveljanik** [comment]
ACK https://github.com/bitcoin/bitcoin/pull/9866/commits/1e9e320dc98a69661484842bfde9d1c24fb52ecb

Maybe we should (in other PR) add a small doc about using Bitcoin Core on low-memory systems (recommendation for IBD, for normal run, blocksonly etc.).

**Comment by MarcoFalke** [comment]
ACK 1e9e320

**Comment by laanwj** [comment]
utACK

> Maybe we should (in other PR) add a small doc about using Bitcoin Core on low-memory systems (recommendation for IBD, for normal run, blocksonly etc.).

I have [such a document](https://gist.github.com/laanwj/efe29c7661ce9b6620a7) though it's not been updated in a while so some things are probably no longer relevant.

**Review by gmaxwell — APPROVED** [review]
ACK.

File: doc/release-notes.md [hunk 1]
```diff
@@ -342,6 +342,16 @@ Fundrawtransaction change address reuse
 - Users should also consider using `getrawchangeaddress()` in conjunction
   with `fundrawtransaction`'s `changeAddress` option.
 
+Unused mempool memory used by coincache
+----------------------------------------
+
+- Before 0.14, memory reserved for mempool (using the `-maxmempool` option)
+  went unused during initial block download. In 0.14, the UTXO DB cache
+  (controled with the `-dbcache` option) borrows memory from the mempool
```
Comment (sdaftuar) [inline doc/release-notes.md]: nit: "controlled"

Comment (TheBlueMatt) [inline doc/release-notes.md]: Fixed.

File: doc/release-notes.md [hunk 2]
```diff
@@ -342,6 +342,16 @@ Fundrawtransaction change address reuse
...
   with `fundrawtransaction`'s `changeAddress` option.
 
+Unused mempool memory used by coincache
+----------------------------------------
+
+- Before 0.14, memory reserved for mempool (using the `-maxmempool` option)
+  went unused during initial block download. In 0.14, the UTXO DB cache
+  (controlled with the `-dbcache` option) borrows memory from the mempool
+  when there is extra memory available. This may result in an increase in
+  memory usage during IBD for those previously relying on only the `-dbcache`
```
Comment (sdaftuar) [inline doc/release-notes.md]: (Sorry for trickling my review comments, but I was about to ACK and then just noticed this:)  Perhaps "IBD" is too much jargon for release notes?

Comment (TheBlueMatt) [inline doc/release-notes.md]: Hmm, "initial block download" was mentioned a few lines up...I clarified there.

**Review by dcousens — APPROVED** [review]

**Comment by sipa** [comment]
ACK.

It may be worth it to also update @laanwj's low-memory suggestions, and include it in the doc/ directory?

**Comment by laanwj** [comment]
> It may be worth it to also update @laanwj's low-memory suggestions, and include it in the doc/ directory?

Yes that would be useful, I planned to do that at some point when the document was ready. But it was never ready because things are always changing :)

**Pull request merged**